app/assets/javascripts/initializers/initializeVideoPlayback.js

```diff
@@ -0,0 +1,193 @@
+      return hms = (+a[0]) * 60 * 60 + (+a[1]) * 60 + (+a[2])
+    }
+  }
+
+  function initWebPlayer(seconds, metadata) {
+    var waitingOnJWP = setInterval(function () {
+      if (typeof jwplayer !== 'undefined') {
+        clearInterval(waitingOnJWP);
+        var playerInstance = jwplayer(`video-player-${metadata.id}`);
+        playerInstance.setup({
+          file: metadata.video_source_url,
+          mediaid: metadata.video_code,
+          autostart: true,
+          image: metadata.video_thumbnail_url,
+          playbackRateControls: true,
+          tracks: [{
+            file: metadata.video_closed_caption_track_url,
+            label: "English",
+            kind: "captions",
+            "default": false
+          }]
+        });
+        if (seconds) {
+          jwplayer().on('ready', function (event) {
+            jwplayer().play();
+          });
+          jwplayer().on('firstFrame', function () {
+            jwplayer().seek(timeToSeconds(seconds))
+          });
+        }
+      }
+    }, 2);
+  }
+
+  function videoMetadata(videoSource) {
+    try {
+      return JSON.parse(videoSource.dataset.meta);
+    } catch (e) {
+      console.log('Unable to load Podcast Episode metadata', e); // eslint-disable-line no-console
+    }
+  }
+
+  function requestFocus() {
+    var metadata = videoMetadata(videoSource);
+    var playerElement = getById(`video-player-${metadata.id}`);
+
+    getById('pause-butt').classList.add('active');
+    getById('play-butt').classList.remove('active');
+
+    nativeBridgeMessage({
+      action: 'play',
+      url: metadata.video_source_url,
+      seconds: currentTime,
+    });
+  }
+
+  function handleVideoMessages(mutation) {
+    if (mutation.type !== 'attributes') {
+      return;
+    }
+
+    var message = {};
+    try {
+      var messageData = getById('video-player-source').dataset.message;
+      message = JSON.parse(messageData);
+    } catch (e) {
+      console.log(e); // eslint-disable-line no-console
+      return;
+    }
+
+    if (message.action == 'pause') {
+      getById('pause-butt').classList.remove('active');
+      getById('play-butt').classList.add('active');
+    } else if (message.action == 'tick') {
+      currentTime = message.currentTime;
+    }
+  }
+
+  function initializePlayer(videoSource) {
+    var seconds = getParameterByName('t') || '0';
+    var metadata = videoMetadata(videoSource);
+
+    if (isNativeIOS()) {
+      nativeBridgeMessage = function(message) {
+        try {
+          window.webkit.messageHandlers.video.postMessage(message);
+        } catch (err) {
+          console.log(err.message); // eslint-disable-line no-console
+        }
+      }
+    } else if(isNativeAndroid()) {
+      nativeBridgeMessage = function(message) {
+        try {
+          AndroidBridge.videoMessage(JSON.stringify(message));
+        } catch (err) {
+          console.log(err.message); // eslint-disable-line no-console
+        }
+      }
```

Comment:
@citizen428 [I remember your suggestion](https://github.com/thepracticaldev/dev.to/pull/8054#discussion_r430210129) from last time I worked on native bridge JS. These checks are performed once and from then on the function assigned to `nativeBridgeMessage` is called directly.

I know in this case the effect is not as impactful as in the Podcast code (much more complex), but let me know what you think about this implementation of the strategy pattern. I think I can squeeze in a refactor to the podcast code in a separate PR to use this and avoid all those _if checks_ before our cooldown is over 🙂

Reply:
@fdoxyz I like it! 😃 As you said yourself, it's probably more impactful for podcasts, but this is a nice trial run for the idea.